### PR TITLE
MM-47902: add tests for onprem_clearbit

### PR DESCRIPTION
**Pull request by ifoukarakis**
#### Summary

- [x] Introduce tests for cloud clearbit.
- [x] Restructure clearbit related fixture files to support cloud and onprem clearbit.
- [x] Extract `expected_data` from `test_cloud_clearbit.py` in order to be reusable in `test_onprem_clerabit.py`. The fixture factory is customized in each file.

Note that there's a few bugs that were discovered in `onprem_clearbit.py`. These bugs shall be fixed in a follow up PR.

Note that the PR is relatively big because:
- Fixtures are specified as json files ( ~ 2/3 of the additions).
- Original file was edited so that code was moved inside a function.

### CODE REVIEW

File: tests/utils/test_onprem_clearbit.py [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+        exception_data,
+        # GIVEN: items to update
+        items_to_update,
+    ]
+
+    # GIVEN: clearbit returns provided reveal responses
+    mock_clearbit.Reveal.find.side_effect = mock_clearbit_reveal(*clearbit_responses)
+
+    # WHEN: request to cloud clearbit
+    onprem_clearbit()
```
Comment (catalintomai) [inline tests/utils/test_onprem_clearbit.py:247]: @ifoukarakis , is this the right cloud clearbit call?

File: tests/utils/test_onprem_clearbit.py [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+        # GIVEN: clearbit exception table contains data? (second request to load dataframe)
+        exception_data,
+        # GIVEN: items to update
+        items_to_update,
+    ]
+
+    # GIVEN: clearbit returns provided reveal responses
+    mock_clearbit.Reveal.find.side_effect = mock_clearbit_reveal(*clearbit_responses)
+
+    # WHEN: request to cloud clearbit
```
Comment (catalintomai) [inline tests/utils/test_onprem_clearbit.py:246]: is it cloud or onprem?

**Review by catalintomai — APPROVED** [review]

**Review by munish7771 — COMMENTED** [review]
Need to update comments in test_onprem_clearbit.

**Review by munish7771 — APPROVED** [review]

File: tests/utils/test_onprem_clearbit.py (hunk 2 above)
Comment (ifoukarakis) [reply tests/utils/test_onprem_clearbit.py:246]: Copy paste error, fixing!

**Pull request merged**